### PR TITLE
feat: generate code examples automatically from tests

### PR DESCRIPTION
**Pull request by yrobla**
Closes: #1622

### CODE REVIEW

**Comment by yrobla** [comment]
image

**Comment by depombo** [comment]
Hard to appreciate from the screenshot, but if the code samples are getting buried down it might make sense to create another top-level section in the TOC called `Sample Queries`. `Reference` can stay as-is and `Sample Queries` can list every module as well, but include the queries only. That way we can also exclude the `Code Samples` title too. Something like this:

```

Sample Queries >
   iasql_functions
   aws_account
Reference >
   iasql_functions
   aws_account

```

More of a thought than a proper suggestion. Curious to hear your opinion @yrobla. It is hard for me to know if this is substantially more difficult or makes the code structure more logical

File: test/modules/aws-codepipeline-integration.ts [hunk 1]
```diff
@@ -709,7 +709,7 @@ describe('AwsCodepipeline Integration Testing', () => {
     try {
       query(
         `
-      UPDATE pipeline_declaration SET region='${nonDefaultRegion} WHERE name='${prefix}-${dbAlias}');
+      UPDATE pipeline_declaration SET region='${nonDefaultRegion}' WHERE name='${prefix}-${dbAlias}');
```
Comment (aguillenv) [inline test/modules/aws-codepipeline-integration.ts:712]: I wonder how was this working 👀

**Comment by yrobla** [comment]
image

**Comment by yrobla** [comment]
> 

i divided into folders, and added collapsible content, as we discussed

File: test/modules/aws-cloudfront-integration.ts [hunk 2]
```diff
@@ -170,7 +170,7 @@ describe('Cloudfront Integration Testing', () => {
       () => ({ username, password }),
     )((e?: any) => {
       if (!!e) return done(e);
-      done();
+      done(undefined);
```
Comment (aguillenv) [inline test/modules/aws-cloudfront-integration.ts:173]: why is this change needed?

**Review by depombo — APPROVED** [review]

File: test/helpers.ts [hunk 3]
```diff
@@ -200,3 +200,7 @@ export function defaultRegion(overrideList?: string[]): string {
   ];
   return regionList[Math.floor(Math.random() * regionList.length)];
 }
+
+export function itDocs(...args: any[]): any {
+  return (it as unknown as (...args: any[]) => {})(...args);
```
Comment (aguillenv) [inline test/helpers.ts:205]: Maybe this solve then issue of the `undefined`s around on the `done` functions?
```suggestion
  return (it as unknown as (...args?: any[]) => {})(...args);
```
(just thinking out loud)

Comment (yrobla) [reply test/helpers.ts:205]: no, i tried already... A rest parameter cannot be optional.ts(1047)

**Review by dfellis — APPROVED** [review]
Hope the new docs work in Dark Mode :)
